core-bundle/src/Routing/RouteProvider.php

```diff
@@ -242,7 +243,15 @@ private function createCollectionForRoutes(array $routes, array $languages): Rou
 
     private function addRoutesForPage(PageModel $page, array &$routes): void
     {
-        $page->loadDetails();
+        try {
+            $page->loadDetails();
+
+            if (!$page->rootId) {
+                throw new NoRootPageFoundException();
```

Comment:
Why not `return;` here?

Reply:
Could do as well. But the exception is actually what happens – no root page was found. Kind of describing the code 😉

Reply:
I find it rather confusing, because I initially thought that you are catching your own exception. Had to look up the method to understand your changes. Not very descriptive IMHO. 🤷‍♂

Reply:
Changed in 121e0cd56de00f7c28136ce5251b923ac3cbd125